src/hooks/auth/useGetUserFeatureFlags.ts

```diff
@@ -0,0 +1,20 @@
+import useSWR from 'swr';
+
+import { getUserFeatureFlags } from '@/utils/auth/api';
+import { makeUserFeatureFlagsUrl } from '@/utils/auth/apiPaths';
+import { isLoggedIn } from '@/utils/auth/login';
+
+const useGetUserFeatureFlags = () => {
+  const { data, isValidating, error } = useSWR(
+    isLoggedIn() ? makeUserFeatureFlagsUrl() : null,
+    getUserFeatureFlags,
+  );
+
+  return {
+    data,
+    isLoading: isValidating && !data,
+    error,
+  };
+};
```

Comment:
I think we can refactor this hook to make it easier to use. For an example, when we use, it check for the feature flag everytime manually. What about accepting the flag in the hook, just return `isEnabled` (will be `false` if not logged in) and `isLoading`, and handle the error as well since it's a passive request so I think the handling will always be the same (whether to just ignore it or maybe show a toast).

So the final result would look like:
```ts
const useGetUserFeatureFlags = (flag: string) => {
  const { data, isValidating, error } = useSWR(
    isLoggedIn() ? makeUserFeatureFlagsUrl() : null,
    getUserFeatureFlags,
  );


  const isLoading = isValidating && !data;
  return {
    isEnabled: isLoggedIn() && !isLoading && !error && data?.[flag],
    isLoading,
  };
};
```

We also shouldn't worry about using the hook many times in different places since SWR will handle deduping them, though I think we might need `useSWRImmutable`.

Reply:
Yes I like the refactor suggestion it's much more reusabled. Just bare in mind that this is temporary until we deploy Unleash :) I will refactor the code. Regarding using `useSWRImmutable`, I think `useSWR` is better incase we have online users and we happen to change their feature flags and we want to reflect the changes as soon as possible (disaster plan for when Novu deployment goes wrong :) )